sound/soc/soc-core.c

```diff
@@ -2148,6 +2167,7 @@ int snd_soc_poweroff(struct device *dev)
 EXPORT_SYMBOL_GPL(snd_soc_poweroff);
 
 const struct dev_pm_ops snd_soc_pm_ops = {
+	.prepare = snd_soc_prepare,
```

Comment:
The `.prepare` is used when the device is suspending, preceding the `.suspend` callback if I recall right.
Are you suspecting that a work from a previous resume is still executing or waiting for execution?
If that is the case then should we try to cancel it if it is not already running (we are after all going to suspend) or wait until it finishes to execute the work (`cancel_work_sync()`)?

Reply:
yes, the traces show that we try to bring the device in D0, but suspend immediately.
This patch was intended to delay the suspend until the resume is actually completed.

However the traces also show that the freeze error happens before we reach the .prepare, so that's not helping.